### DISCUSSION TITLE
Going Fwd with MONAI-Deploy-App-SDK

### DISCUSSION THREAD

**Discussion by vikashg** [Project-MONAI/monai-deploy-app-sdk#489] [Ideas]
In this discussion board I intend to start a conversation about the next features in MONAI-Deploy-App-SDK. Here are two things that I am thinking about 
1. New Operators to load LLM models from hugging-face/OpenAI for generating some language based output. One potential application could be after we generate the image based output. We can also generate some sentences along with the AI result to go into the radiology report.
2. HL7: We are talking about this for almost a year to have a HL7 support in monai-deploy-app-sdk. I think we are lacking a coherent story line where we can show a complete end-to-end pipeline with HL7 support. 
3. Multimodal support: This is somewhat similar to point 1. Once we have the operators for absorbing LLM models, we can use the operator in use case like making a prediction using LLM first then using then chaining the LLM output with some MONAI image based output to present a complete picture. 
4. Maybe adding support for [CDSHooks](https://cds-hooks.org/)
5. Make an example for MAP with minikube or any k8s cluster.

**Comment by vikashg** [comment]
@dbericat @MMelQin for visibility

**Comment by dbericat** [comment]
@hshuaib90

**Comment by hshuaib90** [comment]
I would be a big proponent for (2). Specifically creating a standard ORU HL7 message that developers can use to create standardised HL7 result outputs. The informatics gateway or other HL7 engine could then emit this message to RIS or PACS to either record the result and/or triage the worklist.

**Comment by vikashg** [comment]
I just saw this yesterday. https://github.com/epic-open-source/seismometer
We have been talking about model monitoring and EPIC integration. We were not able to look into this because EPIC was closed. But I feel we should investigate and integrate this with monai deploy. It will be big news if we can do that.  I think this investigation should be our priority.
